### PR TITLE
Refactor PredefinedShelfUtils using static methods

### LINKED ISSUE #436: Refactor PredefinedShelfUtils to have static methods
Currently, `PredefinedShelfUtils` has non-static methods and needs to be instantiated. Refactor this so it cannot be instantiated and all of its methods are static

---

If you need anything, feel free to get in touch with us over [Slack](https://join.slack.com/t/teambookproject/shared_invite/zt-h6vx3n8l-KN8QnO50r7QWZHgHFnFdPw)

### PR DESCRIPTION
**Pull request by daviduster**
## Summary of change

Refactor PredefinedShelfUtils to have static methods.

## Related issue

Closes #436

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [x] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [x] Created a branch that has a descriptive name (what your branch is for in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

If in doubt, get in touch with us via our [Slack workspace](https://join.slack.com/t/teambookproject/shared_invite/zt-h6vx3n8l-KN8QnO50r7QWZHgHFnFdPw)

### CODE REVIEW

**Comment by knjk04** [comment]
As is stands, you won't show up as a contributor. One possible reason for this is because your email address attached to your commits is/are not the same as those associated with your GitHub account. 

In line with the ['Ensure your email address on your commits are correct'](https://github.com/Project-Books/book-project/blob/master/CONTRIBUTING.md#ensure-your-email-address-on-your-commits-are-correct) section our style guide, you'll need to fix this. Instructions on how to do so can be found [here](https://github.com/Project-Books/book-project/blob/master/CONTRIBUTING.md#rewriting-pushed-commits-with-a-different-email-address)

For more information, see https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-profile/why-are-my-contributions-not-showing-up-on-my-profile

I'll mark this PR as a draft. Feel free to request a review when you have fixed this.

If you need a hand with this, let us know over [Slack](https://join.slack.com/t/teambookproject/shared_invite/zt-h6vx3n8l-KN8QnO50r7QWZHgHFnFdPw).

File: src/main/java/com/karankumar/bookproject/backend/utils/PredefinedShelfUtils.java [hunk 1]
```diff
@@ -33,16 +33,19 @@
...
 
 import lombok.extern.java.Log;
+import org.springframework.beans.factory.annotation.Autowired;
 
 @Log
 public class PredefinedShelfUtils {
-    private final PredefinedShelfService predefinedShelfService;
+
+    @Autowired
+    private static PredefinedShelfService predefinedShelfService;
```
Comment (amit1307) [inline src/main/java/com/karankumar/bookproject/backend/utils/PredefinedShelfUtils.java:42]: I don't see any usage of the service in this class so do we need this change?

File: src/main/java/com/karankumar/bookproject/backend/utils/PredefinedShelfUtils.java [hunk 2]
```diff
@@ -72,13 +75,13 @@ public static boolean isPredefinedShelf(String shelfName) {
...
      */
-    public Set<Book> getBooksInChosenPredefinedShelf(String chosenShelf) {
+    public static Set<Book> getBooksInChosenPredefinedShelf(String chosenShelf) {
         Set<Book> books;
         if (isAllBooksShelf(chosenShelf)) {
             return getBooksInAllPredefinedShelves();
         }
 
-        PredefinedShelf.ShelfName predefinedShelfName = getPredefinedShelfName(chosenShelf);
+        PredefinedShelf.ShelfName predefinedShelfName = PredefinedShelfUtils.getPredefinedShelfName(chosenShelf);
```
Comment (amit1307) [inline src/main/java/com/karankumar/bookproject/backend/utils/PredefinedShelfUtils.java:84]: Since we are calling the method with in same class we don't need to use class name?

File: src/main/java/com/karankumar/bookproject/ui/shelf/CustomShelfForm.java [hunk 3]
```diff
@@ -98,8 +98,9 @@ private boolean customShelfNameAlreadyUsed(String customShelfName) {
     }
 
     private boolean customShelfNameMatchesPredefinedShelfName(String shelfName) {
+        new PredefinedShelfUtils(predefinedShelfService);
```
Comment (amit1307) [inline src/main/java/com/karankumar/bookproject/ui/shelf/CustomShelfForm.java:101]: Sorry, didn't get why we need to create an instance here and not using it anywhere?

File: src/test/java/com/karankumar/bookproject/ui/components/utils/ComponentUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import com.vaadin.flow.component.checkbox.Checkbox;
+import com.vaadin.flow.component.combobox.ComboBox;
+import com.vaadin.flow.component.textfield.IntegerField;
+import com.vaadin.flow.component.textfield.TextField;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+class ComponentUtilTest {
```
Comment (amit1307) [inline src/test/java/com/karankumar/bookproject/ui/components/utils/ComponentUtilTest.java:31]: Is this change related to this PR?

**Review by amit1307 — CHANGES_REQUESTED** [review]

File: src/test/java/com/karankumar/bookproject/ui/components/utils/ComponentUtilTest.java (hunk 4 above)
Comment (knjk04) [reply src/test/java/com/karankumar/bookproject/ui/components/utils/ComponentUtilTest.java:31]: @daviduster Perhaps this PR diff is larger than it needs to be because you merged in Vivek's changes (the 'add test for clearComponentField' commit)?

**Comment by knjk04** [comment]
@daviduster How are you getting on with addressing the comments on this pull request? If you need more time, that's fine, but please let us know.

If you need any help on this, feel free to reach out to us on [Slack](https://join.slack.com/t/teambookproject/shared_invite/zt-i7lept44-rgJ9yB0A2vedJTLyyfkjKQ)

**Comment by knjk04** [comment]
@daviduster As we have not heard back from you in a while, in line with our contributing guidelines, I will close this PR and unassign you from the issue
